queries.sql

```diff
@@ -8,3 +8,51 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+/* day 2 project */
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL OR species = '';
+SELECT * FROM animals;
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT my_savepoint;
+UPDATE animals SET weight_kg = weight_kg * -1;
+SELECT * FROM animals;
+ROLLBACK TO my_savepoint;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+SELECT * FROM animals;
+
+SELECT COUNT(*) FROM animals;
+
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) as average_weight
+FROM animals;
+
+SELECT neutered, COUNT(escape_attempts) as escape_attempts_count
+FROM animals
+GROUP BY neutered
+ORDER BY escape_attempts_count DESC;
+
+SELECT name, escape_attempts
+FROM animals
+WHERE escape_attempts = (SELECT MAX(escape_attempts) FROM animals);
+
```

Comment:
Kindly note that the query for the maximum and minimum weight of each animal should be similar to this :

`   SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals
   GROUP BY species;
`

Reply:
Also, the query for  "What is the average number of escape attempts per animal type of those born between 1990 and 2000?" should be similar to this :
`   SELECT species, AVG(escape_attempts) FROM animals
   WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/12/31'
   GROUP BY species;

`

Kindly write the query to produce accurate results 😄